specification/cognitiveservices/ContentSafety/models.tsp

```diff
@@ -228,3 +228,137 @@ model JailbreakAnalysisResult {
   @doc("Analysis result for jailbreak.")
   detected: boolean;
 }
+
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "MUST fix in next update"
+@added(ContentSafety.Versions.v2024_02_15_Preview)
+@doc("Domain.")
+enum Domain {
+  Generic,
+  Medical,
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "MUST fix in next update"
+@added(ContentSafety.Versions.v2024_02_15_Preview)
+@doc("Task type.")
+enum Task {
+  Summarization,
+  QnA,
+}
+
+@added(Versions.v2024_02_15_Preview)
+@doc("Connection details for the GPT resource.")
+model GptResource {
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Property types must use camelCase"
+  @doc("Endpoint for Azure OpenAI resource.")
+  azureOpenAIEndpoint: string;
+
+  @doc("Deployment model name.")
+  deploymentName: string;
+}
+
+@added(Versions.v2024_02_15_Preview)
+@doc("The request of ungroundedness detection.")
+model DetectUngroundednessOptions {
+  @doc("""
+  The domain of the text for analysis. Allowed values: Medical, Generic.
+  This field is optional, with a default value of Generic.
+  """)
+  domain?: Domain;
+
+  @doc("""
+  The task type for the text analysis. Type of task: QnA, Summarization.
+  This field is optional, with a default value of Summarization.
+  """)
+  task?: Task;
+
+  @doc("""
+  The user's question input in a QnA scenario.
+  This field is optional, but if the task type is set to QnA, it becomes required.
+  """)
+  @maxLength(7500)
+  query?: string;
```

Comment:
Interdependent properties quickly become difficult to use. How do you see this evolving over time?

Reply:
From the current perspective, this dependency is necessary. In the Q&A scenario, we require users to provide the query itself as context.